src/nri/sampler.go

```diff
@@ -72,6 +70,15 @@ func (cs *ContainerSampler) SampleAll(ctx context.Context, i *integration.Integr
 	}
 
 	for _, container := range containers {
+		metrics, err := cs.metrics.Process(container.ID)
+		if err != nil {
+			log.Error("error fetching metrics for container %v: %v", container.ID, err)
+			if _, ok := err.(*biz.ErrExitedContainerExpired); ok {
+				log.Debug("skipping exited container older than TTL")
```

Comment:
`ErrExitedContainerExpired` should contain a more descriptive error message (including the container ID), right?
Additionally I think `Info` might be a better level, as end users can also benefit from seeing this.
```suggestion
			if exitError, ok := err.(*biz.ErrExitedContainerExpired); ok {
				log.Info(exitError.Error())
```

Reply:
Just FYI, this might spam logs quite heavily over time with repeated log lines.

The error will contain all the possible information, it's created with:

```go
ErrExitedContainerExpired{fmt.Sprintf("container %s exited after TTL (%v), skipping", containerID, mc.exitedContainerTTL)}

Reply:
Let me create a new suggestion, as I think it is easy to understand each other that way :)